src/avalara/client.py

```diff
@@ -63,11 +75,16 @@ def __init__(self, app_name=None, app_version=None, machine_name=None,
         self.app_name = app_name
         self.app_version = app_version
         self.machine_name = machine_name
-        self.client_id = '{}; {}; Python SDK; API_VERSION; {};'.format(app_name,
-                                                                app_version,
-                                                                machine_name)
+        self.client_id = '{}; {}; Python SDK; API_VERSION; {};'.format(
+            app_name, app_version, machine_name
+        )
         self.client_header = {'X-Avalara-Client': self.client_id}
-        self.timeout_limit = timeout_limit 
+        self.timeout_limit = timeout_limit
+        self.is_log_req_resp_allowed = is_log_req_resp_allowed
+        self.logger = None
```

Comment:
SDK consumer can inject their own logger with config (please refer code example in description of PR).
Since it is a property of class they can set it after instantiation of class however I have added logger as an optional constructor parameter.